packages/flame_fire_atlas/lib/flame_fire_atlas.dart

```diff
@@ -270,24 +271,43 @@ class FireAtlas {
   }
 
   /// Loads the [FireAtlas] from an asset.
+  ///
+  /// Use [encoded] = false to load the asset from a json file.
   static Future<FireAtlas> loadAsset(
     String fileName, {
     AssetsCache? assets,
     Images? images,
+    bool encoded = true,
   }) async {
     final assetsCache = assets ?? Flame.assets;
-
-    final bytes = await assetsCache.readBinaryFile(fileName);
-    final atlas = FireAtlas.deserialize(bytes);
+    final FireAtlas atlas;
+    final fileExtension = path.extension(fileName);
+    if (encoded) {
+      assert(fileExtension == '.fa');
```

Comment:
I feel it is a bit conflicting to require both the specification of the `encoded` parameter *and* at the same time enforcing the file extension.
if we are enforcing the file extension anyway, can't we just use that to determine the format instead of having to specify the parameter?
or, alternatively: if we want the option for the user to fine-control behaviour with the parameter, then shouldn't we allow them to use whatever file name they prefer?

Reply:
I feel like Luan in here, Maybe we don't really need to have that enforcement? Or did you do that for some reason that we are missing @Arenukvern ?

Reply:
I removed the assertions, is there anything more that should be done?